pydarn/plotting/power.py

```diff
@@ -0,0 +1,150 @@
+
+        return "This class provides the following methods: \n"\
+                " - plot_interference()"
+
+    @classmethod
+    def plot_pwr0_statistic(cls, records: List[dict], beam_num: int = 0,
+                            compare: bool = True, frequency: float = 11000,
+                            statistical_calc: object = np.mean):
+
+        """
+        This function will calculate and plot a statistic of the lag-0
+        power of each record as a function of time.
+
+        This function applies the statistical function (ex. numpy.mean)
+        to the pwr0 vector (lag-0 power for each range) for each record
+        before plotting the results from all records chronologically.
+
+        Notes
+        -----
+        This code can be used to study background interference in rawacf data
+        when the radar has been operating in a receive-only mode such as 
+        "politescan" (cpid -3380), or during periods without any obvious 
+        coherent scatter returns from any range.
+
+        If you wish to compare the background interference associated with two
+        different frequencies then let compare=true. The frequencies will be
+        organized by the 'frequency' input. For example, if politescan ran
+        with 10.3 and 12.2 MHz then you could let 'frequency' equal 11000.
+        All records with tfreq below 11000 will be separated from those records
+        with tfreq above 11000. If compare = False then frequency is simply
+        the frequency that politescan ran with i.e 12.2 MHz exclusively.
+
+        Future Work
+        -----------
+        Allow for multi-beam comparison to get a directional
+        sense of the interference
+
+
+        Parameters
+        ----------
+        records: List[dict]
+        beam_num: int
+            The beam number with the desired data to plot
+        compare: bool
+            determines if a single frequency is use in plotting lag0
+            (compare=False) or two frequencies between a frequency
+            (compare=True)
+            default: False
+        frequency: int
+            frequency to specifically look for or split between depending
+            on the setting of compare.
+            default: 1100 kHz
+        statistical_calc: numpy object
+            numpy statistical calculation or generic min or max functions
+            e.g. numpy.std, numpy.median, numpy.min, numpy.max
+            default: numpy.mean
+
+        Raise
+        -----
+        NoDataFound
+        """
+
+        if compare:
+            # now compare frequencies separated by a frequency cutoff
+            # predefine the lists of separated records
+            low_freq_records = [record for record in records
+                                if record['tfreq'] < frequency]
+            high_freq_records = [record for record in records
+                                 if record['tfreq'] > frequency]
+
+            if len(low_freq_records) == 0 and high_freq_records == 0:
```

Comment:
This should be an "or" in order to not cause an error later.
```suggestion
            if len(low_freq_records) == 0 or high_freq_records == 0:
```

Reply:
I think `and` is needed since you could get only frequencies below or higher which is still something to plot.

Reply:
this error will occur if it's not and OR:
>  File "/home/marci/code/pydarn/pydarn/plotting/power.py", line 99, in plot_pwr0_statistic
    stid = high_freq_records[0]['stid']
IndexError: list index out of range

Reply:
actually it's missing the len() call too. 
```suggestion
            if len(low_freq_records) == 0 or len(high_freq_records) == 0:
```

Reply:
if you want to leave as an 'and', I think that the code below might just have to change to move to compare=False if there is only one frequency.